tests/Service/QueryParametersParserTest.php

```diff
@@ -0,0 +1,105 @@
+<?php declare(strict_types=1);
+
+namespace Lmc\ApiFilter\Service;
+
+use Lmc\ApiFilter\AbstractTestCase;
+use Lmc\ApiFilter\Entity\Value;
+use Lmc\ApiFilter\Filter\FilterWithOperator;
+use Lmc\ApiFilter\Filters\Filters;
+use Lmc\ApiFilter\Filters\FiltersInterface;
+
+class QueryParametersParserTest extends AbstractTestCase
+{
+    /** @var QueryParametersParser */
+    private $queryParametersParser;
+
+    protected function setUp(): void
+    {
+        $this->queryParametersParser = new QueryParametersParser();
+    }
+
+    /**
+     * @test
+     * @dataProvider provideQueryParameters
+     */
+    public function shouldParseQueryParameters(array $queryParameters, FiltersInterface $expectedFilters): void
+    {
+        $result = $this->queryParametersParser->parse($queryParameters);
+
+        $this->assertEquals($expectedFilters, $result);
+    }
+
+    public function provideQueryParameters(): array
+    {
+        return [
+            // queryParameters, expectedFilters
+            'empty' => [[], new Filters([])],
+            'simple eq' => [
+                ['title' => 'foo'],
+                Filters::from([new FilterWithOperator('title', new Value('foo'), '=')]),
```

Comment:
Can't you strip the `Filters::from(` par from each dataset, an do it in the test body?